libs/blocks/mas-autoblock/mas-autoblock.js

```diff
@@ -22,32 +33,147 @@ export function getTagName(el) {
+  }
+}
+
+export function getCollectionSidenav(collection) {
+  if (!collection.data) return null;
+  const { hierarchy, placeholders } = collection.data;
+  if (!hierarchy) return null;
+
+  const titleKey = `${collection.variant}SidenavTitle`;
+  const sidenav = createTag('merch-sidenav', { sidenavTitle: placeholders?.[titleKey] || '' });
+
+  /* Search */
+  const searchText = placeholders?.searchText;
+  if (searchText) {
+    const spectrumSearch = createTag('sp-search', { placeholder: searchText });
+    const search = createTag('merch-search', { deeplink: 'search' });
+    search.append(spectrumSearch);
+    sidenav.append(search);
+  }
+
+  /* Filters */
+  const spSidenav = createTag('sp-sidenav', { manageTabIndex: true });
+  spSidenav.setAttribute('manageTabIndex', true);
+  const sidenavList = createTag('merch-sidenav-list', { deeplink: 'filter' }, spSidenav);
+
+  let multilevel = false;
+  function generateLevelItems(level, parent) {
+    for (const node of level) {
+      const value = node.label.toLowerCase();
+      const item = createTag('sp-sidenav-item', { label: node.label, value });
+      if (node.icon) {
+        const icon = createTag('img', { src: node.icon, slot: 'icon', style: 'height: fit-content;' });
+        item.append(icon);
+      }
+      parent.append(item);
+      if (node.collections) {
+        multilevel = true;
+        generateLevelItems(node.collections, item);
+      }
+    }
+  }
+
+  generateLevelItems(hierarchy, spSidenav);
+  if (multilevel) spSidenav.setAttribute('variant', 'multilevel');
+
+  sidenav.append(sidenavList);
+
+  return sidenav;
+}
+
+export async function createMasElement(el, options, tagName) {
+  const [attributes, html] = getTagAttributesAndContent(options.fragment, tagName);
+  const masElement = createTag(tagName, attributes, html);
+  let element = masElement;
+
+  if (tagName === 'merch-card-collection' && options.sidenav) {
+    const container = createTag('div', null, masElement);
+    element = container;
+  }
+
+  el.replaceWith(element);
+  await checkReady(masElement);
+
+  if (tagName === 'merch-card-collection') {
+    /* Placeholders */
+    const placeholders = masElement.data?.placeholders || {};
+    for (const key of Object.keys(placeholders)) {
+      const value = placeholders[key];
+      const tag = value.includes('<p>') ? 'div' : 'p';
+      const placeholder = createTag(tag, { slot: key }, value);
+      masElement.append(placeholder);
+    }
+
+    /* Sidenav */
+    if (options.sidenav) {
+      element.classList.add(`${masElement.variant}-container`);
+      const sidenav = getCollectionSidenav(masElement);
+      if (sidenav) {
+        element.insertBefore(sidenav, masElement);
+        masElement.sidenav = sidenav;
+      }
+    }
+
+    masElement.requestUpdate();
   }
 }
 
 export default async function init(el) {
-  const fragment = getFragmentId(el);
-  if (!fragment) return;
-  await createCard(el, fragment);
+  const options = getOptions(el);
```

Comment:
We need something like to include following PR's change.
https://github.com/adobecom/milo/pull/3873

```suggestion
  const options = getOptions(el);
  options.fragment = options.fragment || options.query
```

Reply:
I actually created a PR for this.
https://github.com/adobecom/milo/pull/3884

Reply:
Yeah, I saw your PR, but didn't know it was already merged. Thanks, fixed